### PR TITLE
Add new design to thermostat card

### PR DESCRIPTION
**Pull request by piitaya**
## Breaking change

Design has changed.

## Proposed change

Add new design to thermostat card inspired from more info. The bottom of the card is configurable using climate preset mode and hvac mode [features](https://www.home-assistant.io/dashboards/tile/#tile-features).
Note : humidifier card will follow in another PR : https://github.com/home-assistant/frontend/pull/18711

CleanShot 2023-11-21 at 10 35 49

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/types.ts [hunk 1]
```diff
@@ -452,6 +452,7 @@ export interface ThermostatCardConfig extends LovelaceCardConfig {
   entity: string;
   theme?: string;
   name?: string;
+  features?: LovelaceTileFeatureConfig[];
```
Comment (bramkragten) [inline src/panels/lovelace/cards/types.ts:455]:
```suggestion
  features?: LovelaceFeatureConfig[];
```

File: src/panels/lovelace/editor/config-elements/hui-thermostat-card-editor.ts [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
 import type { ThermostatCardConfig } from "../../cards/types";
+import {
+  LovelaceTileFeatureConfig,
+  LovelaceTileFeatureContext,
+} from "../../tile-features/types";
 import type { LovelaceCardEditor } from "../../types";
+import "../hui-sub-element-editor";
 import { baseLovelaceCardConfig } from "../structs/base-card-struct";
+import { EditSubElementEvent, SubElementEditorConfig } from "../types";
+import "./hui-tile-card-features-editor";
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-thermostat-card-editor.ts:26]:
```suggestion
import "./hui-features-editor";
```

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 3]
```diff
@@ -296,223 +96,56 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
...
-        tabindex="0"
-        .path=${modeIcons[mode]}
-        .label=${this.hass!.localize(
-          `component.climate.entity_component._.state.${mode}`
-        ) || mode}
-      >
-      </ha-icon-button>
+      <ha-card>
+        <p class="title">${name}</p>
+        <ha-more-info-climate-temperature
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-thermostat-card.ts:122]:
```suggestion
        <ha-climate-control
```

Comment (piitaya) [reply src/panels/lovelace/cards/hui-thermostat-card.ts:122]: I was planning to do this in a future PR but ok, let's do this now.

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-thermostat-card.ts:122]: future PR is fine too

**Review by bramkragten — APPROVED** [review]

**Pull request merged**